### PR TITLE
Fix DBP test case 

### PR DESCRIPTION
**Pull request by quanganhdo**
Task/Issue URL: https://app.asana.com/0/72649045549333/1208757448779624/f
Tech Design URL:
CC:

**Description**:

Fixes a test case to ensure when a profile is removed, scheduled opt-out request is cancelled.

**Steps to test this PR**:
1. CI passes

**Definition of Done**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)

### CODE REVIEW

File: LocalPackages/DataBrokerProtection/Tests/DataBrokerProtectionTests/DataBrokerProfileQueryOperationManagerTests.swift [hunk 1]
```diff
@@ -389,6 +396,8 @@ final class DataBrokerProfileQueryOperationManagerTests: XCTestCase {
             XCTAssertTrue(mockDatabase.optOutEvents.contains(where: { $0.type == .optOutConfirmed }))
             XCTAssertTrue(mockDatabase.wasUpdateRemoveDateCalled)
             XCTAssertNotNil(mockDatabase.extractedProfileRemovedDate)
+            XCTAssertTrue(mockDatabase.wasUpdatedPreferredRunDateForOptOutCalled)
```
Comment (quanganhdo) [inline LocalPackages/DataBrokerProtection/Tests/DataBrokerProtectionTests/DataBrokerProfileQueryOperationManagerTests.swift:399]: This assertion disappeared some time in the last year. Added it back to check for the correct preferred run date.

Comment (THISISDINOSAUR) [reply LocalPackages/DataBrokerProtection/Tests/DataBrokerProtectionTests/DataBrokerProfileQueryOperationManagerTests.swift:399]: I'm worried this was removed deliberately. I do remember at one point the deliberate choice to keep doing certain operations for old data even when a user had changed their details.

Comment (quanganhdo) [reply LocalPackages/DataBrokerProtection/Tests/DataBrokerProtectionTests/DataBrokerProfileQueryOperationManagerTests.swift:399]: @jotaemepereira/@Bunn Do you have any context on this?

Comment (jotaemepereira) [reply LocalPackages/DataBrokerProtection/Tests/DataBrokerProtectionTests/DataBrokerProfileQueryOperationManagerTests.swift:399]: @quanganhdo, I do not have the full context of the change; I need to dig deeper. What @THISISDINOSAUR is mentioning is true.

When a user changes their profile and there are current opt-outs in progress for the old profile, we will:
- Not scan for new matches in the old profile
- In-progress matches for the old profile should be driven to completion (opt-out and then confirmation)

I do not know if this helps; ping me on MM if you need more assistance.

**Pull request closed without merging**